services/main-frontend/src/pages/manage/organizations/[id]/index.tsx

```diff
@@ -1,61 +1,831 @@
+  const [activeTab, setActiveTab] = React.useState<TabKey>(DEFAULT_TAB)
+  const [showAddUserPopup, setShowAddUserPopup] = React.useState(false)
+  const [editMode, setEditMode] = React.useState(false)
+  const [users, setUsers] = React.useState<NamedRoleUser[]>([])
+  const [email, setEmail] = React.useState("")
+  const [role, setRole] = React.useState("")
+  const [hidden, setHidden] = React.useState(false)
+  const [showEditPopup, setShowEditPopup] = React.useState(false)
+  const [editUser, setEditUser] = React.useState<RoleUser | null>(null)
+  const [editRole, setEditRole] = React.useState("")
+  const [editedName, setEditedName] = React.useState("")
+  const [showDeletePopup, setShowDeletePopup] = React.useState(false)
+  const [editedSlug, setEditedSlug] = React.useState("")
+
+  const router = useRouter()
+
+  const addMutation = useToastMutation(
+    () => {
+      // eslint-disable-next-line i18next/no-literal-string
+      return giveRole(email, role as UserRole, { tag: "Organization", id: query.id })
+    },
+    { notify: true, method: "POST" },
+    {
+      onSuccess: () => {
+        setEmail("")
+        setRole("")
+        setShowAddUserPopup(false)
+        roleQuery.refetch()
+      },
+    },
+  )
+
+  const deleteMutation = useToastMutation(
+    async () => {
+      const res = await fetch(`/api/v0/main-frontend/organizations/${query.id}`, {
+        method: "PATCH", // keep PATCH here — this is the real HTTP method
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          deleted_at: new Date().toISOString(),
+        }),
+      })
+
+      if (!res.ok) {
+        let json
+        try {
+          json = await res.json()
+        } catch {
+          throw new Error(`Soft delete failed: ${res.status}`)
+        }
+        throw json
+      }
+
+      return res
+    },
+    { notify: true, method: "PUT" },
+    {
+      onSuccess: () => {
+        router.push(allOrganizationsRoute())
+      },
+    },
+  )
+
+  const handleDelete = (userToDelete: NamedRoleUser) => {
+    if (window.confirm(t("confirm-delete-user", { email: userToDelete.email }))) {
+      // eslint-disable-next-line i18next/no-literal-string
+      removeRole(userToDelete.email, userToDelete.role, { tag: "Organization", id: query.id })
+        .then(() => {
+          roleQuery.refetch()
+        })
+        .catch((error) => {
+          console.error("Failed to delete user role:", error)
+        })
+    }
+  }
+
+  const handleEdit = (userToEdit: NamedRoleUser) => {
+    setEditUser(userToEdit)
+    setEditRole(userToEdit.role)
+    setShowEditPopup(true)
+  }
+
+  const handleSaveEdit = () => {
+    if (!editUser || !editRole || editRole === editUser.role) {
+      setShowEditPopup(false)
+      return
+    }
+
+    // eslint-disable-next-line i18next/no-literal-string
+    void removeRole(editUser.email, editUser.role, { tag: "Organization", id: query.id })
+      .then(() =>
```

Comment:
Never use then or catch when you can make the function async and use await and try-catch. That makes programming so much easier.